### PR TITLE
PA-5569 Documentation updates

### PR DESCRIPTION
**Pull request by MichaelL-PA**
https://soos.atlassian.net/browse/PA-5569

- [x] update the helptext for parameters to be as through as possible, providing a link to an external source if need be

- [x] make sure that --help can be used

- [x] find an easy way to convert the information from --help into a markdown table that can be put into the README (this needs to be either relatively easy or copy-paste-able)

Use --helpFormatted

- [x] provide instructions in the PR templates or README for how to keep that table up-to-date

### CODE REVIEW

File: src/cli/soos.py [hunk 1]
```diff
@@ -1759,6 +1774,19 @@ def entry_point():
         else:
             sys.exit(0)
 
+def print_help_formatted(parser):
+    print("| Argument | Description |")
+    print("| --- | --- |")
+
+    allRows = []
+    for arg, helpText in parser._option_string_actions.items():
+        descriptionText = helpText.help.replace('\n', '<br>')
```
Comment (SOOS-JAlvarez) [inline src/cli/soos.py:1783]: would be nice if the default value is included
`defaultValue = helpText.default`

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<!---
+If you've edited any of the arguments for this package:
+
+1. Run this script with the --helpFormatted argument (ex. soos-sca --helpFormatted)
+2. Copy the result and paste it in the README under '### Script Arguments'
+3. Make sure your terminal didn't wrap any lines, confirm the table looks correct
+-->
```
Comment (SOOS-JAlvarez) [inline .github/pull_request_template.md:7]:
```suggestion
### Changes:
  - Description of the change

**Ticket:** https://soos.atlassian.net/browse/PA-0000

<!---
If you've edited any of the arguments for this package:

1. Run this script with the --helpFormatted argument (ex. soos-dast --helpFormatted)
2. Copy the result and paste it in the README under '### Script Arguments'
3. Make sure your terminal didn't wrap any lines, confirm the table looks correct
-->
```

Taking the oportunity add these

File: src/cli/soos.py (hunk 1 above)
Comment (gregjoeval) [reply src/cli/soos.py:1783]: yea if we could add a column for default values that would be great

File: .github/pull_request_template.md (hunk 2 above)
Comment (MichaelL-PA) [reply .github/pull_request_template.md:7]: good point

File: src/cli/soos.py (hunk 1 above)
Comment (MichaelL-PA) [reply src/cli/soos.py]: good idea

File: src/cli/soos.py [hunk 3]
```diff
@@ -1380,28 +1381,37 @@ def register_arguments():
...
                             default="run_and_wait",
                             required=False
                             )
 
-        parser.add_argument("--onFailure", "-of", dest="on_failure",
-                            help="On Failure: "
-                                 "fail_the_build: Fail The Build "
+        parser.add_argument("-of", "--onFailure", dest="on_failure",
+                            help="On Failure:\n"
+                                 "fail_the_build: Fail The Build\n"
```
[the remark this answers is not in the document]
Comment (gregjoeval) [reply src/cli/soos.py:1407]: might be be necessary for the `-h` command

**Review by SOOS-JAlvarez — APPROVED** [review]

File: src/cli/soos.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (MichaelL-PA) [reply src/cli/soos.py:1407]: think we do.

it doesn't show up on the terminal I was using, for the readme table it's nice to have though
```
On Failure:
fail_the_build: Fail The Build
continue_on_failure: Continue On Failure ** Default Value
```
vs 

```
On Failure: fail_the_build: Fail The Build continue_on_failure: Continue On Failure ** Default Value
```

[the remark this answers is not in the document]
Comment (gregjoeval) [reply src/cli/soos.py:1407]: what i meant was to add it automatically

**Pull request merged**